crates/rslint_parser/src/syntax/stmt.rs

```diff
@@ -576,24 +585,45 @@ pub(crate) fn directives(p: &mut Parser) -> Option<ParserState> {
 		// eat semicolon if present, correct termination guaranteed by is_directive
 		p.eat(SyntaxKind::SEMICOLON);
 
-		directive.complete(p, JS_DIRECTIVE);
+		let completed_marker = directive.complete(p, JS_DIRECTIVE);
 
 		let directive_text = p.token_src(&directive_token);
 
 		if directive_text == "\"use strict\"" || directive_text == "'use strict'" {
-			if old_state == None {
-				old_state = Some(p.state.clone());
+			if self.old_state == None {
+				self.old_state = Some(p.state.clone());
 			}
 
 			let mut new_state = p.state.clone();
 			new_state.strict(p, directive_token.range);
 			p.state = new_state;
 		}
+
+		Present(completed_marker)
+	}
+
+	fn is_at_list_end(&mut self, p: &mut Parser) -> bool {
+		!self.at_directives(p)
 	}
 
-	list.complete(p, LIST);
+	fn recover(
+		&mut self,
+		p: &mut Parser,
+		parsed_element: ParsedSyntax<Self::ParsedElement>,
+	) -> RecoveryResult {
+		parsed_element.or_recover(
+			p,
+			&ParseRecovery::new(JS_UNKNOWN_STATEMENT, STMT_RECOVERY_SET),
+			js_parse_error::expected_directive,
+		)
+	}
+}
```

Comment:
I'm not sure if it makes sense to use `ParseList` for directives because we won't need any error recovery for directives (we only parse as long as it's a string literal... if not, we abort. 
```suggestion
	fn recover(
		&mut self,
		p: &mut Parser,
		parsed_element: ParsedSyntax<Self::ParsedElement>,
	) -> RecoveryResult {
		Err(RecoveryError::AlreadyRecovered)
	}
}
```